### DISCUSSION TITLE
A port with only headers to be used for loading the library dynamically

### DISCUSSION THREAD

**Discussion by asfernandes** [microsoft/vcpkg#42195] [General]
In https://github.com/microsoft/vcpkg/pull/36176 I have tried to integrate Firebird client library but due to a number of issues abandoned for now.

But it's also common to use the Firebird API dynamically, i.e., together with LoadLibrary/dlopen.

And I'm writing a library that just does that and I think I should not include Firebird headers in my library.

So I would want to know if a Firebird port only with the headers would be accepted, and if yes, should it be the same name that in future may have a build or it should use a special name.

PS: I'm a Firebird committer

**Comment by autoantwort** [comment]
> So I would want to know if a Firebird port only with the headers would be accepted

This seems useless if you still need the build the project outside of vcpkg to use it.

**Comment by asfernandes** [comment]
The firebird client is always a dynamic library. Usually, applications do not distribute it and let the user choose what the one installed will be used.

So no, this is not useless. And BTW, the modern API is a single function that returns a struct and additional method are get from it. It's a perfect valid usage and would enable the deployment of additional libraries in vcpkg, as the firebird one is much lower level one.
